test_util/cluster_api.py

```diff
@@ -201,55 +202,50 @@ def __init__(self, dcos_uri, masters, public_masters, slaves, public_slaves,
 
         # URI must include scheme
         assert dcos_uri.startswith('http')
-        self.scheme = urlparse(dcos_uri).scheme
+        parse_result = urlparse(dcos_uri)
+        self.scheme = parse_result.scheme
+        self.dns_host = parse_result.netloc.split(':')[0]
 
         # Make URI never end with /
         self.dcos_uri = dcos_uri.rstrip('/')
 
-    @staticmethod
-    def _marathon_req_headers():
-        return {'Accept': 'application/json, text/plain, */*'}
+        # Generate request functions
+        for method in self.request_methods:
```

Comment:
Note: one reason I chose this somewhat awkward pattern is that in downstream tests, we need to bind verify=my_ca_cert_path for all these same methods and its nice to do it in a programmatic manner to prevent divergence. Case in point: in the current method with the manually wrapped request methods, PUT takes in **kwargs and never passes it and also breaks the parity with requests lib by mapping the json keyword to payload instead.